dialogflow/entity_type_management.rb

```diff
@@ -0,0 +1,121 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def list_entity_types project_id:
+  # [START dialogflow_list_entity_types]
+  # project_id = "Your Google Cloud project ID"
+  
+  require "google/cloud/dialogflow"
+
+  entity_types_client = Google::Cloud::Dialogflow::EntityTypes.new
+  parent = entity_types_client.class.project_agent_path project_id
+
+  entity_types = entity_types_client.list_entity_types(parent)
+
+  entity_types.each do |entity_type|
+    puts "Entity type name:         #{entity_type.name}"
+    puts "Entity type display name: #{entity_type.display_name}"
+    puts "Number of entities:       #{entity_type.entities.size}"
+  end
+  # [END dialogflow_list_entity_types]
+end
+
+
+def create_entity_type project_id:, display_name:, kind:
+  # [START dialogflow_create_entity_type]
+  # project_id = "Your Google Cloud project ID"
+  # display_name = "some_display_name"
+  # kind = :KIND_LIST
+  
+  require "google/cloud/dialogflow"
+
+  entity_types_client = Google::Cloud::Dialogflow::EntityTypes.new
+  parent = entity_types_client.class.project_agent_path project_id
+
+  entity_type = { display_name: display_name, kind: kind }
+  response = entity_types_client.create_entity_type parent, entity_type
+
+  puts "Entity type created: #{response}"
+  puts "Display name:        #{response.display_name}"
+  # [END dialogflow_create_entity_type]
+end
+
+
+def delete_entity_type project_id:, entity_type_id:
+  # [START dialogflow_delete_entity_type]
+  # project_id = "Your Google Cloud project ID"
+  # entity_type_id = "some_entity_type_id"
+  
+  require "google/cloud/dialogflow"
+
+  entity_types_client = Google::Cloud::Dialogflow::EntityTypes.new
+  entity_type_path = entity_types_client.class.entity_type_path project_id, entity_type_id
+
+  response = entity_types_client.delete_entity_type entity_type_path
+  # [END dialogflow_delete_entity_type]
+end
+
+
+# Helper
+def get_entity_type_ids project_id:, display_name:
+  # project_id = "Your Google Cloud project ID"
+  # display_name = "some_display_name"
+  
+  require "google/cloud/dialogflow"
+
+  entity_types_client = Google::Cloud::Dialogflow::EntityTypes.new
+  parent = entity_types_client.class.project_agent_path project_id
```

Comment:
Instead of using `.class`, write out the constant name, eg. `Google::Cloud::Dialogflow::SomethingService.project_agent_path`

Resist the Ruby temptation to use Reflection 💎